packages/thirdweb/src/extensions/erc4337/account/addSessionKey.ts

```diff
@@ -86,3 +91,84 @@ export function addSessionKey(
 export function isAddSessionKeySupported(availableSelectors: string[]) {
   return isSetPermissionsForSignerSupported(availableSelectors);
 }
+
+/**
+ * Checks if the session key should be updated.
+ * @param currentPermissions - The current permissions of the session key.
+ * @param newPermissions - The new permissions to set for the session key.
+ * @returns A boolean indicating if the session key should be updated.
+ */
+export async function shouldUpdateSessionKey(args: {
+  accountContract: ThirdwebContract;
+  sessionKeyAddress: string;
+  newPermissions: AccountPermissions;
+}): Promise<boolean> {
+  const { accountContract, sessionKeyAddress, newPermissions } = args;
+
+  // check if account is deployed
+  const accountDeployed = await isContractDeployed(accountContract);
+  if (!accountDeployed) {
+    return true;
+  }
+
+  // get current permissions
+  const currentPermissions = await getPermissionsForSigner({
+    contract: accountContract,
+    signer: sessionKeyAddress,
+  });
+  // check end time validity
+  if (
+    currentPermissions.endTimestamp &&
+    currentPermissions.endTimestamp < Math.floor(new Date().getTime() / 1000)
+  ) {
+    return true;
+  }
+
+  // check targets
+  if (
+    !areSessionKeyContractTargetsEqual(
+      currentPermissions.approvedTargets,
+      newPermissions.approvedTargets,
+    )
+  ) {
+    return true;
+  }
+
+  // check native token limit
+  if (
+    currentPermissions.nativeTokenLimitPerTransaction !==
+    toWei(newPermissions.nativeTokenLimitPerTransaction?.toString() ?? "0")
```

Comment:
Again instead of equality, this could be `currentPermissions.nativeTokenLimitPerTransaction >= newPermissions.nativeTokenLimitPerTransaction`

Reply:
done